public/components/filter_bar/filter_editor.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCodeEditor,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiPopoverTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { i18n } from '@osd/i18n';
+
+interface Props {
+  content: string;
+  label: string;
+  onSubmit: (content: string, label: string) => void;
+  onCancel: () => void;
+}
+
+const isFilterValid = (content: string) => {
+  try {
+    return Boolean(JSON.parse(content));
+  } catch (e) {
+    return false;
```

Comment:
Shall we add a error notification to user it it has the error?

Reply:
Error is already provided by editor. This validation is required to disable save button if there is an error.

![Uploading Screenshot 2023-04-04 at 1.37.08 AM.png…]()

Reply:
In the video, If I put mouse on top of editor page and scroll the mouse scroll wheel , mostly it couldn’t scroll down and up, but if I put mouse on top of scroll bar, it behaves normally.
https://user-images.githubusercontent.com/90288540/229680704-544b9845-b9a9-4793-8d6b-e7775153084a.mov